### PR TITLE
[CBRD-24470] improved ddl audit

### PR DESCRIPTION
**Pull request by ctshim**
http://jira.cubrid.org/browse/CBRD-24470

* Remove unnecessary memory allocation and deallocation.
* Make sure that the query string is copied only if it contains the DDL syntax.

### CODE REVIEW

File: src/base/ddl_log.c [hunk 1]
```diff
@@ -153,306 +170,346 @@ logddl_free (bool all_free)
...
 
 bool
 logddl_get_jsp_mode ()
 {
   bool jsp_mode = false;
-  if (ddl_audit_handle)
-    {
-      jsp_mode = ddl_audit_handle->jsp_mode;
-    }
+  //if (ddl_audit_handle)
```
Comment (beyondykk9) [inline src/base/ddl_log.c]: supposed to be removed?

Comment (ctshim) [reply src/base/ddl_log.c]: OK, I will do that.

File: src/base/ddl_log.c [hunk 2]
```diff
@@ -58,25 +58,33 @@
...
+#define SQL_TEXT_SIZE               (2048)
+#define SQL_TEXT_SIZE_MAX           (8192)
+#define SQL_TEXT_INCR               (1024)
 
 typedef struct t_ddl_audit_handle T_DDL_AUDIT_HANDLE;
 struct t_ddl_audit_handle
 {
+  char *sql_text;
+  int alloc_size;		// Allocated size of sql_text  
+  char zsql_text[SQL_TEXT_SIZE];
```
Comment (hyunikn) [inline src/base/ddl_log.c:70]: It should be SQL_TEXT_SIZE + 1 to avoid buffer overflow at line 328.

File: src/base/ddl_log.c [hunk 3]
```diff
@@ -153,306 +170,341 @@ logddl_free (bool all_free)
...
-  ddl_audit_handle->stmt_type = -1;
-  ddl_audit_handle->execute_type = LOGDDL_RUN_EXECUTE_FUNC;
-  ddl_audit_handle->loaddb_file_type = LOADDB_FILE_TYPE_NONE;
-  ddl_audit_handle->log_filepath[0] = '\0';
-  ddl_audit_handle->str_qry_exec_begin_time[0] = '\0';
-  ddl_audit_handle->elapsed_time[0] = '\0';
-  ddl_audit_handle->file_line_number = 0;
-  ddl_audit_handle->err_code = 0;
-  ddl_audit_handle->msg[0] = '\0';
+  if (ddl_audit_handle.alloc_size > SQL_TEXT_SIZE_MAX)
```
Comment (hyunikn) [inline src/base/ddl_log.c:174]: Possibly, you meant SQL_TEXT_SIZE, not SQL_TEXT_SIZE_MAX. 
Or, the condition can be "ddl_audit_handle.sql_text != ddl_audit_handle.zsql_text" as in line 222.

File: src/base/ddl_log.c (hunk 2 above)
Comment (ctshim) [reply src/base/ddl_log.c:70]: There is no danger in this part.
It is guaranteed that the buffer is assigned to a larger size than the ddl_audit_handle.alloc_size on line 328.

File: src/base/ddl_log.c (hunk 3 above)
Comment (ctshim) [reply src/base/ddl_log.c:174]: Dynamically allocated memory continues to be reused by default.
However, if the size is larger than SQL_TEXT_SIZE_MAX, it is reverting to its initial state.
The final memory release is at logddl_destroy().

Comment (hyunikn) [reply src/base/ddl_log.c:174]: OK. I see.

File: src/base/ddl_log.c (hunk 2 above)
Comment (hyunikn) [reply src/base/ddl_log.c:70]: It seems possible that "ddl_audit_handle.sql_text == ddl_audit_handle.zsql_text" at line 328.
In that case, the size of the buffer is not larger than but is equal to ddl_audit_handle.alloc_size.

Comment (ctshim) [reply src/base/ddl_log.c:70]: I understand.
I'll fix it.

**Review by hyunikn — APPROVED** [review]

File: src/broker/cas_function.c [hunk 4]
```diff
@@ -2742,17 +2747,17 @@ update_error_query_count (T_APPL_SERVER_INFO * as_info_p, const T_ERROR_INFO * e
...
 {
-  char stmt_type = -1;
   for (int i = 0; i < srv_handle->num_q_result; i++)
     {
       if (logddl_is_ddl_type (srv_handle->q_result[i].stmt_type) == true)
 	{
-	  stmt_type = srv_handle->q_result[i].stmt_type;
-	  break;
+	  logddl_set_stmt_type (srv_handle->q_result[i].stmt_type);
+	  return;
```
Comment (hgryoo) [inline src/broker/cas_function.c:2762]: the function name does not explain the routine. how about renaming the function?
```
logddl_check_ddl ()...
```

**Review by beyondykk9 — APPROVED** [review]

File: src/broker/cas_function.c (hunk 4 above)
Comment (ctshim) [reply src/broker/cas_function.c:2762]: Changed to logddl_check_have_ddl_stmt().
Thank you.

**Review by hgryoo — APPROVED** [review]

**Review by shparkcubrid — APPROVED** [review]

**Review by youngjinj — APPROVED** [review]

**Review by airnet73 — APPROVED** [review]

**Pull request merged**